lib/models/Cat.js

```diff
@@ -0,0 +1,50 @@
+const pool = require('../utils/pool');
+
+module.exports = class Cat {
+  id;
+  name;
+  favoriteToy;
+
+  constructor(row) {
+    this.id = row.id;
+    this.name = row.name;
+    this.favoriteToy = row.favorite_toy;
+  }
+
+  static async insert({ name, favoriteToy }) {
+    const { rows } = await pool.query(
+      'INSERT INTO cats (name, favorite_toy) VALUES ($1, $2) RETURNING *;',
+      [name, favoriteToy]
+    );
```

Comment:
We may want to add an early return after all of these SQL queries to prevent errors in the case of nothing being returned from the databse. something like `if(!rows[0] return null`